src/kvtree_util.c

```diff
@@ -225,3 +225,149 @@ int kvtree_util_get_ptr(const kvtree* hash, const char* key, void** val)
 
   return rc;
 }
+
+/*
+ * Read the contents of a summary file (rank2file) and all of it's subfiles
+ * (rank2file.x.y) into a single big kvtree.  This is similiar to
+ * kvtree_read_scatter() but non-MPI.
+ *
+ * prefix:  The directory path containing the rank2file* files.
+ * data:    A new, empty, kvtree that gets filled in by this function.
+ *
+ * Returns KVTREE_SUCCESS on success, non-KVTREE_SUCCESS otherwise.
+ */
+int kvtree_read_scatter_single(const char* prefix, kvtree* data)
+{
+  char path[PATH_MAX];
+  kvtree* subfile_kvtree;
+  int rc;
+  kvtree_elem* elem;
+  kvtree* final_rank_kvtree;
+
+  /* Read in the contents of the main 'rank2file' tree. */
+  snprintf(path, sizeof(path), "%s/rank2file", prefix);
+  path[sizeof(path) - 1] = '\0';
+
+  rc = kvtree_read_file(path, data);
+  if (rc != KVTREE_SUCCESS) {
+    return rc;
+  }
+
+  /* Break off and remove the "RANK" subtree from the rank2file tree */
+  subfile_kvtree = kvtree_extract(data, "RANK");
+  if (!subfile_kvtree) {
+    return KVTREE_FAILURE;
+  }
+
+  /*
+   * subfile_kvtree looks like:
+   *
+   * 55
+   *   FILE
+   *     .0.55
+   *   OFFSET
+   *     0
+   * 27
+   *   OFFSET
+   *     0
+   *   FILE
+   *     .0.27
+   * 0
+   *   FILE
+   *     .0.0
+   *   OFFSET
+   *     0
+   */
+
+  /*
+   * final_rank_kvtree will hold merged contents of all the "RANK" subtrees from
+   * each of our subfiles.
+   */
+  final_rank_kvtree = kvtree_new();
+
+  /* Iterate though all the RANKS filename */
+  for (elem = kvtree_elem_first(subfile_kvtree); elem != NULL;
+      elem = kvtree_elem_next(elem))
+  {
+    /*
+     * For each one of these groups in "RANKS" tree:
+     *  55
+     *   FILE
+     *     .0.55
+     *   OFFSET
+     *     0
+     *
+     * key = 55, subfile = .0.55, offset = 0
+     *
+     * Use this info to reconstruct the subfile name.
+     */
+    const char* key = kvtree_elem_key(elem);
+    char* subfile;
+    char* offset_str;
+    off_t offset;
+    kvtree* local_config = kvtree_elem_hash(elem);
+    kvtree* subtree = kvtree_new();
+    kvtree* ranks_subtree;
+
+    if (kvtree_util_get_str(local_config, "FILE", &subfile) != KVTREE_SUCCESS) {
+        return KVTREE_FAILURE;
+    }
+    if (kvtree_util_get_str(local_config, "OFFSET", &offset_str) != KVTREE_SUCCESS) {
+        return KVTREE_FAILURE;
+    }
+    offset = strtoul(offset_str, NULL, 10);
```

Comment:
We use kvtree_util_set_bytecount to set this value, so let's use kvtree_util_get_bytecount to read it back.